beluga/test/benchmark/benchmark_raycasting.cpp

```diff
@@ -50,12 +50,124 @@ BENCHMARK_CAPTURE(BM_Bresenham2i, Modified, beluga::Bresenham2i::kModified)
 
 using beluga::testing::StaticOccupancyGrid;
 
+template <std::size_t Rows, std::size_t Cols>
+class BaselineGrid : public beluga::BaseOccupancyGrid2<BaselineGrid<Rows, Cols>> {
+ public:
+  struct ValueTraits {
+    [[nodiscard]] bool is_free(bool value) const { return !value; }
+    [[nodiscard]] bool is_unknown(bool) const { return false; }
+    [[nodiscard]] bool is_occupied(bool value) const { return value; }
+  };
+
+  explicit BaselineGrid(std::initializer_list<bool>, double resolution) : resolution_{resolution} {
+    std::fill(std::begin(data()), std::end(data()), false);
+  }
+
+  [[nodiscard]] const Sophus::SE2d& origin() const { return origin_; }
+
+  [[nodiscard]] auto& data() { return grid_; }
+  [[nodiscard]] const auto& data() const { return grid_; }
+  [[nodiscard]] std::size_t size() const { return grid_.size(); }
+
+  [[nodiscard]] std::size_t width() const { return Cols; }
+  [[nodiscard]] std::size_t height() const { return Rows; }
+  [[nodiscard]] double resolution() const { return resolution_; }
+
+  [[nodiscard]] auto value_traits() const { return ValueTraits{}; }
+
+ private:
+  double resolution_;
+  Sophus::SE2d origin_;
+  std::array<bool, Rows * Cols> grid_;
```

Comment:
Question: What's the difference between `BaselineGrid` and `StaticOccupancyGrid`?

Answer: It's the order of members :exploding_head:.

```
BM_RayCasting2d_BaselineRaycast<BaselineGrid>/1024               470 ns          469 ns      1490708
BM_RayCasting2d_BaselineRaycast<StaticOccupancyGrid>/1024       1625 ns         1624 ns       427804
```

Reply:
Can you reproduce this results locally? This is crazy stuff.

Reply:
Wat :exploding_head:

Reply:
I had to try once to believe it, but I see the same.

Poor memory memory alignment for the data members after the array?

### No changes

![Screenshot from 2023-05-29 14-33-06_baseline](https://github.com/Ekumen-OS/beluga/assets/17802342/4e6a4046-1385-4f9b-9fd2-e143ad6cae77)

### Move BaseLineGrid::grid_ to line 79

![Screenshot from 2023-05-29 14-38-34_Baseline grid moved](https://github.com/Ekumen-OS/beluga/assets/17802342/8bb89034-ec0a-4a0e-8b51-e0a33144907d)

Reply:
> Poor memory memory alignment for the data members after the array?

Maybe! It could also be poor cache locality due to the layout of the object in memory.